### PR TITLE
DBAAS-1069: upgrade API and support RDS Aurora

### PR DESCRIPTION
**Pull request by xieshenzh**
Upgrade the operator to use DBaaS API v1beta1, and support importing Aurora and clusters from RDS.

Jira: https://issues.redhat.com/browse/DBAAS-981
https://issues.redhat.com/browse/DBAAS-1069

### CODE REVIEW

**Comment by xieshenzh** [comment]
Based on PR: https://github.com/RHEcosystemAppEng/dbaas-operator/pull/307 and https://github.com/RHEcosystemAppEng/rds-dbaas-operator/pull/15

File: controllers/dbaasprovider_controller.go [hunk 1]
```diff
@@ -177,7 +177,7 @@ func bridgeProviderCR(clusterRoleList *rbac.ClusterRoleList) *dbaasoperator.DBaa
 		},
 
 		Spec: dbaasoperator.DBaaSProviderSpec{
-			Provider: dbaasoperator.DatabaseProvider{
+			Provider: dbaasoperator.DatabaseProviderInfo{
```
Comment (jianrongzhang89) [inline controllers/dbaasprovider_controller.go]: Add GroupVersion to indicate that RDS is using v1beta1 of dbaas API/data structure?

File: Makefile [hunk 2]
```diff
@@ -1,20 +1,23 @@
 CONTAINER_ENGINE?=docker
 
+OPM_VERSION ?= v1.23.2
```
Comment (redhatHameed) [inline Makefile]: @tchughesiv @jeremyary  do we required to update the OPM version for RDS ?

File: bundle/manifests/dbaas.redhat.com_rdsconnections.yaml [hunk 3]
```diff
@@ -34,12 +34,12 @@ spec:
           spec:
             description: Defines the desired state of a DBaaSConnection object.
             properties:
-              instanceID:
```
Comment (jianrongzhang89) [inline bundle/manifests/dbaas.redhat.com_rdsconnections.yaml:37]: If there are existing rdsconnection CRs based on the old schema, we may have issues on upgrading RDS operator to the new release. Have you tested such upgrade? If so could you include the test case and the result in the PR description?

**Review by jeremyary — DISMISSED** [review]

File: bundle/manifests/dbaas.redhat.com_rdsconnections.yaml (hunk 3 above)
Comment (xieshenzh) [reply bundle/manifests/dbaas.redhat.com_rdsconnections.yaml:37]: Yes, I have tested that.
The DBaaS operator will reconcile the DBaaSConnection object, as a result, the spec of the corresponding RDSConnection object will be updated. In most cases, it happens before the rds-dbaas-operator is upgraded and running.

1. I installed RHODA 0.0.3 and created DBaaSInventory, DBaaSInstance and DBaaSConnection objects in the cluster.
2. Then change the CatalogSource of the dbaas-operator to use the 0.0.4 image I built. Then RHODA will upgrade.
Once the upgrade is done and everything is stable, the spec and status of the RDSConnection object will be updated to use v1beta1 spec of DBaaS. And there is no error to reconcile the updated object.
3. Then I created DBaaSInventory, DBaaSInstance and DBaaSConnection objects in the upgraded cluster.
Everything looks fine.

Comment (xieshenzh) [reply bundle/manifests/dbaas.redhat.com_rdsconnections.yaml:37]: What is the specific upgrade issue you are referring to?
I didn't see issue to upgrade the rds-dbaas-operator with these changes.

File: controllers/rdsconnection_controller.go [hunk 4]
```diff
@@ -95,7 +98,14 @@ func (r *RDSConnectionReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 
 	var connection rdsdbaasv1alpha1.RDSConnection
 	var inventory rdsdbaasv1alpha1.RDSInventory
-	var dbInstance rdsv1alpha1.DBInstance
+	var dbService client.Object
+
+	var password *ackv1alpha1.SecretKeyReference
```
Comment (redhatHameed) [inline controllers/rdsconnection_controller.go]: rename to passwordSecret WDYT ?

Comment (xieshenzh) [reply controllers/rdsconnection_controller.go]: Sure, done.

**Review by redhatHameed — APPROVED** [review]
However the testing with UI still remaining will test with one round to make sure it works as expected, if there is any issue we will open the followup PR.

File: bundle/manifests/dbaas.redhat.com_rdsconnections.yaml (hunk 3 above)
Comment (jianrongzhang89) [reply bundle/manifests/dbaas.redhat.com_rdsconnections.yaml:37]: OK

**Pull request merged**